src/protagonist/DLCS.Model/Spaces/ISpaceRepository.cs

```diff
@@ -12,6 +12,23 @@ public interface ISpaceRepository
     
     Task<Space?> GetSpace(int customerId, int spaceId, bool noCache, CancellationToken cancellationToken);
     
+    /// <summary>
+    /// Retrieves a space by name
+    /// </summary>
+    /// <param name="customerId">The customer to retrieve a space from</param>
+    /// <param name="name">The name of the space</param>
+    /// <param name="noCache">Whether the value should be cached</param>
```

Comment:
Results are always cached, `noCache` indicates whether the query can be satisfied by using a cached value or if it should always be fresh from db.